### PR TITLE
EXSWHTEC-262 - Introduce common utilities for atomic arithmetic operations

### PR DESCRIPTION
**Pull request by mirza-halilcevic**
Depends-on: #180

### CODE REVIEW

**Review by scchan — CHANGES_REQUESTED** [review]
Please provide documentation about this test and its algorithms.

**Comment by scchan** [comment]
Please rebase once #180 has been merged.

File: catch/unit/atomics/arithmetic_common.hh [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+#pragma once
+
+#include <hip_test_common.hh>
+#include <hip/hip_cooperative_groups.h>
+#include <resource_guards.hh>
+#include <cmd_options.hh>
+
+namespace cg = cooperative_groups;
+
+enum class AtomicOperation {
```
Comment (b-sumner) [inline catch/unit/atomics/arithmetic_common.hh:33]: This seems to be missing Max and Min and their unsafe variants.

**Comment by mirza-halilcevic** [comment]
> Please provide documentation about this test and its algorithms.

I added some comments to arithmetic_common.hh explaining what each part is responsible for. For a more general overview of what the tests are supposed to do you can check the individual test case descriptions, for example [here](https://github.com/ROCm-Developer-Tools/hip-tests/blob/c3df1a4041efc3c076dbdc27c466c63b91798ce3/catch/unit/atomics/atomicAdd.cc#L35).

File: catch/unit/atomics/arithmetic_common.hh (hunk 1 above)
Comment (mirza-halilcevic) [reply catch/unit/atomics/arithmetic_common.hh:33]: Those are located [here](https://github.com/ROCm-Developer-Tools/hip-tests/pull/192).

Comment (b-sumner) [reply catch/unit/atomics/arithmetic_common.hh:33]: I'm not sure I see the value in having AtomicOperation in different name spaces vs just having a single AtomicOperation with all of the operations, but at least for consistency shouldn't this use an Arithmetic namespace?

**Comment by rakesroy** [comment]
PR has been merged into develop branch via commit 8ee015d1ef00869536b2c14b7f312912753273f1.

**Pull request closed without merging**